### ISSUE TITLE
[BUG] something wrong with `TabulateFusionSeRGradGrad`

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#1681]
### Bug summary

Error throw with `TabulateFusionSeRGradGrad`, when compress training a `se_r` model.

`se_a` does not have this issue.

### DeePMD-kit Version

devel with #1680 applied

### TensorFlow Version

2.7

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

```
Traceback (most recent call last):
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1380, in _do_call
    return fn(*args)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1363, in _run_fn
    return self._call_tf_sessionrun(options, feed_dict, fetch_list,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1456, in _call_tf_sessionrun
    return tf_session.TF_SessionRun_wrapper(self._session, options, feed_dict,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Dim of input should be 3
	 [[{{node gradients_1/gradients/filter_type_0/TabulateFusionSeR_grad/TabulateFusionSeRGrad_1_grad/TabulateFusionSeRGradGrad}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jz748/anaconda3/envs/dpdev113/bin/dp", line 33, in <module>
    sys.exit(load_entry_point('deepmd-kit', 'console_scripts', 'dp')())
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/main.py", line 473, in main
    train_dp(**dict_args)
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 106, in train
    _do_work(jdata, run_opt, is_compress)
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 167, in _do_work
    model.train(train_data, valid_data)
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 514, in train
    run_sess(self.sess, [self.train_op], feed_dict=train_feed_dict,
  File "/home/jz748/codes/deepmd-kit/deepmd/utils/sess.py", line 21, in run_sess
    return sess.run(*args, **kwargs)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 970, in run
    result = self._run(None, fetches, feed_dict, options_ptr,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1193, in _run
    results = self._do_run(handle, final_targets, final_fetches,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1373, in _do_run
    return self._do_call(_run_fn, feeds, fetches, targets, options,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/client/session.py", line 1399, in _do_call
    raise type(e)(node_def, op, message)  # pylint: disable=no-value-for-parameter
tensorflow.python.framework.errors_impl.InvalidArgumentError: Dim of input should be 3
	 [[node gradients_1/gradients/filter_type_0/TabulateFusionSeR_grad/TabulateFusionSeRGrad_1_grad/TabulateFusionSeRGradGrad
 (defined at <string>:4273)
]]

Errors may have originated from an input operation.
Input Source operations connected to node gradients_1/gradients/filter_type_0/TabulateFusionSeR_grad/TabulateFusionSeRGrad_1_grad/TabulateFusionSeRGradGrad:
In[0] Const (defined at /home/jz748/codes/deepmd-kit/deepmd/model/ener.py:278)	
In[1] filter_type_0/TabulateFusionSeR/table_info:	
In[2] filter_type_0/Slice:	
In[3] gradients_1/gradients/filter_type_0/Slice_grad/Pad_1_grad/Slice_1 (defined at /home/jz748/codes/deepmd-kit/deepmd/train/trainer.py:381)	
In[4] filter_type_0/TabulateFusionSeR:

Operation defined at: (most recent call last)
>>>   File "/home/jz748/anaconda3/envs/dpdev113/bin/dp", line 33, in <module>
>>>     sys.exit(load_entry_point('deepmd-kit', 'console_scripts', 'dp')())
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/main.py", line 473, in main
>>>     train_dp(**dict_args)
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 106, in train
>>>     _do_work(jdata, run_opt, is_compress)
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 162, in _do_work
>>>     model.build(train_data, stop_batch)
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 317, in build
>>>     self._build_training()
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 381, in _build_training
>>>     apply_op = optimizer.minimize(loss=self.l2_l,
>>> 
>>>   File "/home/jz748/codes/deepmd-kit/deepmd/op/_tabulate_grad.py", line 40, in _tabulate_fusion_se_r_grad_grad_cc
>>>     dz_dy = op_module.tabulate_fusion_se_r_grad_grad(op.inputs[0], op.inputs[1], op.inputs[2], dy, op.inputs[4])
>>> 
>>>   File "<string>", line 4273, in tabulate_fusion_se_r_grad_grad
>>> 

Original stack trace for 'gradients_1/gradients/filter_type_0/TabulateFusionSeR_grad/TabulateFusionSeRGrad_1_grad/TabulateFusionSeRGradGrad':
  File "/home/jz748/anaconda3/envs/dpdev113/bin/dp", line 33, in <module>
    sys.exit(load_entry_point('deepmd-kit', 'console_scripts', 'dp')())
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/main.py", line 473, in main
    train_dp(**dict_args)
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 106, in train
    _do_work(jdata, run_opt, is_compress)
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 162, in _do_work
    model.build(train_data, stop_batch)
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 317, in build
    self._build_training()
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 381, in _build_training
    apply_op = optimizer.minimize(loss=self.l2_l,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/training/optimizer.py", line 407, in minimize
    grads_and_vars = self.compute_gradients(
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/training/optimizer.py", line 516, in compute_gradients
    grads = gradients.gradients(
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_impl.py", line 169, in gradients
    return gradients_util._GradientsHelper(
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_util.py", line 682, in _GradientsHelper
    in_grads = _MaybeCompile(grad_scope, op, func_call,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_util.py", line 331, in _MaybeCompile
    return grad_fn()  # Exit early
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_util.py", line 683, in <lambda>
    lambda: grad_fn(op, *out_grads))
  File "/home/jz748/codes/deepmd-kit/deepmd/op/_tabulate_grad.py", line 40, in _tabulate_fusion_se_r_grad_grad_cc
    dz_dy = op_module.tabulate_fusion_se_r_grad_grad(op.inputs[0], op.inputs[1], op.inputs[2], dy, op.inputs[4])
  File "<string>", line 4273, in tabulate_fusion_se_r_grad_grad
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/op_def_library.py", line 744, in _apply_op_helper
    op = g._create_op_internal(op_type_name, inputs, dtypes=None,
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 3698, in _create_op_internal
    ret = Operation(
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 2101, in __init__
    self._traceback = tf_stack.extract_stack_for_node(self._c_op)

...which was originally created as op 'gradients/filter_type_0/TabulateFusionSeR_grad/TabulateFusionSeRGrad_1', defined at:
  File "/home/jz748/anaconda3/envs/dpdev113/bin/dp", line 33, in <module>
    sys.exit(load_entry_point('deepmd-kit', 'console_scripts', 'dp')())
[elided 2 identical lines from previous traceback]
  File "/home/jz748/codes/deepmd-kit/deepmd/entrypoints/train.py", line 162, in _do_work
    model.build(train_data, stop_batch)
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 316, in build
    self._build_network(data)
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 340, in _build_network
    = self.model.build (self.place_holders['coord'],
  File "/home/jz748/codes/deepmd-kit/deepmd/model/ener.py", line 235, in build
    = self.descrpt.prod_force_virial (atom_ener, natoms)
  File "/home/jz748/codes/deepmd-kit/deepmd/descriptor/se_r.py", line 416, in prod_force_virial
    [net_deriv] = tf.gradients (atom_ener, self.descrpt_reshape)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_impl.py", line 169, in gradients
    return gradients_util._GradientsHelper(
[elided 2 identical lines from previous traceback]
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/ops/gradients_util.py", line 683, in <lambda>
    lambda: grad_fn(op, *out_grads))
  File "/home/jz748/codes/deepmd-kit/deepmd/op/_tabulate_grad.py", line 35, in _tabulate_fusion_se_r_grad_cc
    dy_df = op_module.tabulate_fusion_se_r_grad(op.inputs[0], op.inputs[1], op.inputs[2], dy, op.outputs[0])
  File "<string>", line 4176, in tabulate_fusion_se_r_grad
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/op_def_library.py", line 744, in _apply_op_helper
    op = g._create_op_internal(op_type_name, inputs, dtypes=None,

...which was originally created as op 'filter_type_0/TabulateFusionSeR', defined at:
  File "/home/jz748/anaconda3/envs/dpdev113/bin/dp", line 33, in <module>
    sys.exit(load_entry_point('deepmd-kit', 'console_scripts', 'dp')())
[elided 4 identical lines from previous traceback]
  File "/home/jz748/codes/deepmd-kit/deepmd/train/trainer.py", line 340, in _build_network
    = self.model.build (self.place_holders['coord'],
  File "/home/jz748/codes/deepmd-kit/deepmd/model/ener.py", line 183, in build
    = self._import_graph_def_from_frz_model(frz_model, feed_dict, return_elements)
  File "/home/jz748/codes/deepmd-kit/deepmd/model/ener.py", line 278, in _import_graph_def_from_frz_model
    return tf.import_graph_def(graph_def, input_map = feed_dict, return_elements = return_elements, name = "")
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/util/deprecation.py", line 552, in new_func
    return func(*args, **kwargs)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/importer.py", line 407, in import_graph_def
    return _import_graph_def_internal(
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/importer.py", line 520, in _import_graph_def_internal
    _ProcessNewOps(graph)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/importer.py", line 251, in _ProcessNewOps
    for new_op in graph._add_new_tf_operations(compute_devices=False):  # pylint: disable=protected-access
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 3848, in _add_new_tf_operations
    new_ops = [
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 3849, in <listcomp>
    self._create_op_from_tf_operation(c_op, compute_device=compute_devices)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 3731, in _create_op_from_tf_operation
    ret = Operation(c_op, self)
  File "/home/jz748/anaconda3/envs/dpdev113/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 2101, in __init__
    self._traceback = tf_stack.extract_stack_for_node(self._c_op)
```

### Steps to Reproduce

```sh
cd examples/water/se_e2_r
dp train input.json
dp freeze
dp compress
dp train input.json -f frozen_model_compressed.pb
```

### Further Information, Files, and Links

The error is throw by https://github.com/deepmodeling/deepmd-kit/blob/92230a3a5f78bb863c19a63f85a9a46dfc1fab88/source/op/tabulate_multi_device.cc#L669

**Comment by njzjz** [comment]
@liangadam take a look?

**Comment by denghuilu** [comment]
I can not reproduce the error with the latest devel code.
```
(tensorflow_venv) root@iZ8vb672g6x8vab6ywhddcZ:~/deepmd-kit/examples/water/se_e2_r# git log | head
commit 087ae56db853bdd59c41a6417eadd7cfdf153f7a
Author: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date:   Thu May 5 21:34:31 2022 -0400

    supports dp convert-from 0.12 (#1685)

    Resolves #1583.

commit e9b27a72049d3628ed7c22ce8e00490c0768b313
Author: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
(tensorflow_venv) root@iZ8vb672g6x8vab6ywhddcZ:~/deepmd-kit/examples/water/se_e2_r# git branch
* devel
(tensorflow_venv) root@iZ8vb672g6x8vab6ywhddcZ:~/deepmd-kit/examples/water/se_e2_r# dp train input.json -f frozen_model.pb
WARNING:tensorflow:From /root/DP-DEBUG/tensorflow_venv/lib/python3.9/site-packages/tensorflow/python/compat/v2_compat.py:111: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
WARNING:deepmd.train.run_options:Switch to serial execution due to lack of horovod module.
/root/DP-DEBUG/tensorflow_venv/lib/python3.9/site-packages/deepmd/utils/compat.py:346: UserWarning: The argument training->numb_test has been deprecated since v2.0.0. Use training->validation_data->batch_size instead.
  warnings.warn(
DEEPMD INFO    Calculate neighbor statistics... (add --skip-neighbor-stat to skip this step)
DEEPMD INFO    training data with min nbor dist: 0.8854385688525511
DEEPMD INFO    training data with max nbor size: [38, 72]
DEEPMD INFO     _____               _____   __  __  _____           _     _  _
DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |
DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_
DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_
DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
DEEPMD INFO    Please read and cite:
DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
DEEPMD INFO    installed to:         /root/deepmd-kit/_skbuild/linux-x86_64-3.9/cmake-install
DEEPMD INFO    source :              v2.1.0-50-g087ae56-dirty
DEEPMD INFO    source brach:         devel
DEEPMD INFO    source commit:        087ae56
DEEPMD INFO    source commit at:     2022-05-06 09:34:31 +0800
DEEPMD INFO    build float prec:     double
DEEPMD INFO    build with tf inc:    /root/DP-DEBUG/tensorflow_venv/lib/python3.9/site-packages/tensorflow/include
DEEPMD INFO    build with tf lib:
DEEPMD INFO    ---Summary of the training---------------------------------------
DEEPMD INFO    running on:           iZ8vb672g6x8vab6ywhddcZ
DEEPMD INFO    computing device:     gpu:0
DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
DEEPMD INFO    Count of visible GPU: 1
DEEPMD INFO    num_intra_threads:    0
DEEPMD INFO    num_inter_threads:    0
DEEPMD INFO    -----------------------------------------------------------------
DEEPMD INFO    ---Summary of DataSystem: training     -----------------------------------------------
DEEPMD INFO    found 3 system(s):
DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
DEEPMD INFO                               ../data/data_0/     192       1      80  0.250    T
DEEPMD INFO                               ../data/data_1/     192       1     160  0.500    T
DEEPMD INFO                               ../data/data_2/     192       1      80  0.250    T
DEEPMD INFO    --------------------------------------------------------------------------------------
DEEPMD INFO    ---Summary of DataSystem: validation   -----------------------------------------------
DEEPMD INFO    found 1 system(s):
DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
DEEPMD INFO                                ../data/data_3     192       1      80  1.000    T
DEEPMD INFO    --------------------------------------------------------------------------------------
DEEPMD INFO    training without frame parameter
DEEPMD INFO    built lr
DEEPMD INFO    built network
DEEPMD INFO    built training
DEEPMD WARNING To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
DEEPMD INFO    initialize training from the frozen model
DEEPMD INFO    start training at lr 5.00e-03 (== 5.00e-03), decay_step 5000, decay_rate 0.942392, final lr will be 3.51e-08
DEEPMD INFO    batch     100 training time 3.01 s, testing time 0.03 s
DEEPMD INFO    batch     200 training time 1.90 s, testing time 0.03 s
DEEPMD INFO    batch     300 training time 1.90 s, testing time 0.03 s
DEEPMD INFO    batch     400 training time 1.90 s, testing time 0.03 s
DEEPMD INFO    batch     500 training time 1.90 s, testing time 0.03 s
```

**Comment by denghuilu** [comment]
#1686
